cloud/services/storage/keypair.go

```diff
@@ -0,0 +1,98 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+
+	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+	tag "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	osc "github.com/outscale/osc-sdk-go/v2"
+)
+
+type OscKeyPairInterface interface {
+	CreateKeyPair(spec *infrastructurev1beta1.OscKeypair, keypairName string) (*osc.KeypairCreated, error)
+	DeleteKeyPair(keypairName string) error
+	ReadKeyPair(keypairName string, keypairFingerprints string) (*osc.Volume, error)
+}
+
+func (s *Service) CreateKeyPair(spec *infrastructurev1beta1.OscKeypair, keypairName string) (*osc.KeypairCreated, error) {
+	keyPairName := spec.Name
+	publicKey := spec.PublicKey
```

Comment:
Is publicKey mandatory for you and why ?
Because it is not in api. (https://docs.outscale.com/api#createkeypair)